examples/camera-app/linux/include/camera-device.h

```diff
@@ -33,12 +33,29 @@
 #define VIDEO_STREAM_GST_DEST_PORT 5000
 #define AUDIO_STREAM_GST_DEST_PORT 5001
 
-#define MAX_CONTENT_BUFFER_SIZE_BYTES (1024)
-#define MAX_ENCODED_PIXEL_RATE (10000)
-#define MAX_CONCURRENT_VIDEO_ENCODERS (1)
-#define MAX_NETWORK_BANDWIDTH_MBPS (64)
-#define MICROPHONE_MIN_LEVEL (1)
-#define MICROPHONE_MAX_LEVEL (254)
+// Camera Constraints set to typical values.
+// TODO: Look into ways to fetch from hardware, if required/possible.
+static constexpr uint32_t kMaxContentBufferSizeBytes = 4096;
+static constexpr uint32_t kMaxNetworkBandwidthMbps   = 128;
+static constexpr uint8_t kMaxConcurrentEncoders      = 1;
+static constexpr uint32_t kMaxEncodedPixelRate       = 27648000; // 720p at 30fps
+static constexpr uint8_t kMicrophoneMinLevel         = 1;
+static constexpr uint8_t kMicrophoneMaxLevel         = 254;  // Spec constraint
+static constexpr uint8_t kMicrophoneMaxChannelCount  = 8;    // Spec Constraint in AudioStreamAllocate
```

Comment:
The min resolution here is a different aspect ratio to the max resolution. Which will cause issues with Digital viewport settings, where you have to maintain the AR.  Max is 16:9, can we set min to 16:9 as well (I used 144p in the stream in camera-device).

Reply:
Oversight. Changed to maintain a 16:9 aspect ratio.